### ISSUE TITLE
Support bucket notifications coming through SNS -> SQS

### ISSUE THREAD

**Issue by jszwedko** [vectordotdev/vector#8185]
A [user in discord](https://discord.com/channels/742820443487993987/753400475545501836/862725634774138890) ran into an issue where they were using the `aws_s3` source with an SQS queue that was receiving bucket notifications via SNS as these have a slightly different format and so Vector rejects them.

I think we could support this message format as well. It seems to just be a wrapper around the normal SQS bucket notification format:

```json
{
  "Type": "Notification",
  "MessageId": "5308912a-6007-5016-850b-c84a03361c61",
  "TopicArn": "arn:aws:sns:us-east-1:428668805589:CloudFlare-Logs",
  "Subject": "Amazon S3 Notification",
  "Message": "{"Records":[{"eventVersion":"2.1","eventSource":"aws:s3","awsRegion":"us-east-1","eventTime":"2021-07-08T02:20:31.059Z","eventName":"ObjectCreated:Put","userIdentity":{"principalId":"AWS:AIDAIDO4SORI5ZZM2GYE2"},"requestParameters":{"sourceIPAddress":"172.70.5.137"},"responseElements":{"x-amz-request-id":"0VKYA15537ZTXDDR","x-amz-id-2":"4grbeCkSZXSGEIqUOY+uOvfYHcKl4mLJkjuOKotyg2+A5YrG+zgScNKkcDOwEYZ+WhNPBag6Qa+O752lS43jHex6q8F6xyq8"},"s3":{"s3SchemaVersion":"1.0","configurationId":"NotifyForGraylogExport","bucket":{"name":"gs-cloudflare","ownerIdentity":{"principalId":"AF4MXOIW3KH3"},"arn":"arn:aws:s3:::gs-cloudflare"},"object":{"key":"gz/20210708T022001Z_20210708T022031Z_4638c872.log.gz","size":115941,"eTag":"caa8c56c98a847e34465de572ac1a13a","sequencer":"0060E660F00FA9D282"}}}]}",
  "Timestamp": "2021-07-08T02:20:33.858Z",
  "SignatureVersion": "1",
  "Signature": "Of9+KXz5vkAMDCNSdTnRH2fgsIF18XMh6KaJheBPzMEbr1lYa0oV9aFHFwu/Z00A54yEgXIczkkHOjQfpsax2RNd5IyRDebHrmRDNUUXkcmUQgOHwND05f2MvUFDLKPmh45KgHiO2PhC8TsuD4NnybBCALp2queXutBR4WBU4w1EpormKbfy+cFrGYLZy4OafK2kphn8ev1J/iDCwEa+smBQ2QNeVDUXKqkjSL1uB61cmTLTWA9aQegsrLU84sxm9bFpF/KjvWpEPiT3X4vdvbwcFU9rfGCOd3YLsKc2PbxONyQNLc+WQL2iaDfEDueFJOXTJHZQ1xQEZX4R6KgkfA==",
  "SigningCertURL": "https://sns.us-east-1.amazonaws.com/SimpleNotificationService-010a507c1833636cd94bdb98bd93083a.pem",
  "UnsubscribeURL": "https://sns.us-east-1.amazonaws.com/?Action=Unsubscribe&SubscriptionArn=arn:aws:sns:us-east-1:428668805589:CloudFlare-Logs:529b6e72-1782-4833-9fc5-06c0a8ac8901"
}
```

**Comment by sHesl** [comment]
:+1: for this feature, but there is also workaround available as well. You can configure SNS topics to maintain the original structure of the S3 notification by setting [raw message delivery](https://docs.aws.amazon.com/sns/latest/dg/sns-large-payload-raw-message-delivery.html) to `true`. 

Had no trouble with an `S3 => SNS fanout => SQS => Vector` pipeline after finding that out. 

Hope someone finds this useful.

**Comment by efiShtain** [comment]
can I work on this issue?

**Comment by spencergilbert** [comment]
> can I work on this issue?

Definitely! Contributions are always welcome, feel free to open a draft and ask questions if you run into anything.

**Comment by sbalmos** [comment]
Depending on what update notification lands where, I've implemented this to a Just Enough To Work For Me state, available for review in #17352

**Comment by jszwedko** [comment]
Closed by https://github.com/vectordotdev/vector/pull/17352
